From https://github.com/hpcng/warewulf/pull/906 Mon Sep 17 00:00:00 2001
From: Matt Jolly <Matt.Jolly@footclan.ninja>
Date: Thu, 7 Sep 2023 18:52:34 +1000
Subject: [PATCH] add error handling to dynamic network configuration

This PR fixes a bug where warewulf would panic if the system had no network connection.

It still fails, but at least now we get an error instead of a backtrace.

diff --git a/internal/pkg/config/root.go b/internal/pkg/config/root.go
index 67c4447af..3e641e6a6 100644
--- a/internal/pkg/config/root.go
+++ b/internal/pkg/config/root.go
@@ -119,7 +119,10 @@ func (conf *RootConf) SetDynamicDefaults() (err error) {
 
 		if conf.Ipaddr == "" {
 			wwlog.Verbose("Configuration has no valid network, going to dynamic values")
-			conn, _ := net.Dial("udp", "8.8.8.8:80")
+			conn, err := net.Dial("udp", "8.8.8.8:80")
+			if err != nil {
+				return errors.Wrap(err, "Couldn't establish a connection to remote")
+			}
 			defer conn.Close()
 			ipaddr = conn.LocalAddr().(*net.UDPAddr).IP
 			mask = ipaddr.DefaultMask()
